imagepipeline/src/main/java/com/facebook/imagepipeline/core/ImagePipeline.java

```diff
@@ -435,7 +452,13 @@ public boolean isInDiskCacheSync(final Uri uri) {
    */
   public boolean isInDiskCacheSync(final ImageRequest imageRequest) {
     final CacheKey cacheKey = mCacheKeyFactory.getEncodedCacheKey(imageRequest, null);
-    return mMainBufferedDiskCache.diskCheckSync(cacheKey);
+    final ImageRequest.CacheChoice cacheChoice = imageRequest.getCacheChoice();
+
+    if (cacheChoice == ImageRequest.CacheChoice.SMALL) {
```

Comment:
style: Let's make this a `switch`. If we ever happen to add something to `CacheChoice`, the compiler will warn us if we forget about this location.